sdk/storage/azure-storage-queue/azure/storage/queue/_models.py

```diff
@@ -275,9 +277,20 @@ def __init__(self, command, results_per_page=None, continuation_token=None):
         )
         self._command = command
         self.results_per_page = results_per_page
+        self.max_messages = max_messages
 
     def _get_next_cb(self, continuation_token):
         try:
+            if self.max_messages is not None:
+                if self.results_per_page is None:
+                    self.results_per_page = 1
+                if self.max_messages == 0:
+                    raise StopIteration("End of paging")
+                if self.results_per_page > self.max_messages:
+                    self.results_per_page = self.max_messages
+                    self.max_messages = 0
+                else:
+                    self.max_messages = self.max_messages - self.results_per_page
```

Comment:
We should probably only set this value after we have successfully retrieved a page, and deduct the number of items that we actually retrieved rather than the number of items we hope to retrieve.